File: Sort_Colors/sort.go
package sortcolors

func sortColors(nums []int) {
	bucket := make([]int, 3)

	for _, num := range nums {
		bucket[num] += 1
	}

	i := 0
	for j := range bucket {
		for bucket[j] > 0 {
			nums[i] = j
			bucket[j]--
			i++
		}
	}
}

File: Sort_Colors/sort.js
/**
 * @param {number[]} nums - The array of number need to sort
 * @returns {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function (nums) {
	const bucket = new Array(3).fill(0);

	for (let i = 0; i < nums.length; i++) {
		bucket[nums[i]]++;
	}

	let pointer = 0;
	for (let i = 0; i < bucket.length; i++) {
		while (bucket[i] > 0) {
			nums[pointer] = i;
			pointer++;
			bucket[i]--;
		}
	}
};
